File: Arrays/P14 Count Maximum Consecutive One's in the array/solution.py
from typing import List

class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        count = 0 
        maxc = 0 
        for i in range(len(nums)):
            if (nums[i] == 1):
                count += 1
                maxc = max(maxc, count)
            else:
                count = 0
                
        
        return maxc

if __name__ == "__main__":
    sol = Solution()
    print(sol.findMaxConsecutiveOnes([1,1,1,1,1,1,1,0,1,1,1]))
    print(sol.findMaxConsecutiveOnes([1,0,1,1,0,1]))  

File: Arrays/P14 Count Maximum Consecutive One's in the array/solution.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int count = 0;
        int maxc = 0;
        
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] == 1) {
                count++;
                maxc = max(maxc, count);
            } else {
                count = 0;
            }
        }
        
        return maxc;
    }
};

int main() {
    Solution sol;
    
    vector<int> nums1 = {1,1,1,1,1,1,1,0,1,1,1};
    vector<int> nums2 = {1,0,1,1,0,1};
    
    cout << sol.findMaxConsecutiveOnes(nums1) << endl;
    cout << sol.findMaxConsecutiveOnes(nums2) << endl;
    
    return 0;
}